lib/portal-builder.js

```
@@ -0,0 +1,126 @@
+const path        = require('path'),
+      _           = require('lodash'),
+      Q           = require('q'),
+      knetikcloud = require('knetikcloud-sdk');
+
+class PortalBuilder {
+
+    /**
+     *
+     * @param {Object} config
+     * @param {string} [config.basePath]
+     * @param {string} config.clientId
+     * @param {string} config.clientSecret
+     */
+    constructor(config) {
+        if(!_.isString(config.clientId))
+            throw new Error('clientId is required');
+        if(!_.isString(config.clientSecret))
+            throw new Error('clientSecret is required');
+
+        this.config = config;
+
+        if(!_.isString(this.config.basePath)) {
+            this.config.basePath = 'https://iot.devsandbox.knetikcloud.com';
```

There should not be a hard coded fallback. This should throw an error.